### PR TITLE
Add NFT metadata

### PR DESCRIPTION
**Pull request by xavikh**

### CODE REVIEW

File: src/escrow/increasing/Lock.sol [hunk 1]
```diff
@@ -38,7 +49,7 @@ contract Lock is ILock, ERC721Enumerable, UUPSUpgradeable, DaoAuthorizable, Reen
 
     function supportsInterface(
         bytes4 _interfaceId
-    ) public view override(ERC721Enumerable) returns (bool) {
+    ) public view override(ERC721Enumerable, ERC721URIStorageUpgradeable) returns (bool) {
```
Comment (jordaniza) [inline src/escrow/increasing/Lock.sol:52]: Strictly speaking, we should test that the supports interface check is correctly returning as expected for Enumerable vs URI Storage. Of the 2 I think Enumerable is more important to be able to introspect so would want to make sure that is not lost through the override

**Review by jordaniza — COMMENTED** [review]
Looks good

File: src/escrow/increasing/Lock.sol (hunk 1 above)
Comment (xavikh) [reply src/escrow/increasing/Lock.sol:52]: True. I added 2 asserts on `test/escrow/escrow/Lock.t.sol` to verify both interfaces are supported. I also checked and all the parent functions use `|| super.supportsInterface(..)` so they should propagate correctly through the inheritance chain.

File: src/escrow/increasing/Lock.sol [hunk 2]
```diff
@@ -2,14 +2,23 @@
...
 
 /// @title NFT representation of an escrow locking mechanism
-contract Lock is ILock, ERC721Enumerable, UUPSUpgradeable, DaoAuthorizable, ReentrancyGuard {
+contract Lock is
+    ILock,
+    ERC721Enumerable,
+    ERC721URIStorageUpgradeable,
+    UUPSUpgradeable,
+    DaoAuthorizable,
+    ReentrancyGuard
```
Comment (jordaniza) [inline src/escrow/increasing/Lock.sol]: inheritance order matters - keep the same order to not mess with storage

**Pull request merged**